### PR TITLE
SNOR-40

### PR DESCRIPTION
**Pull request by dwitek**
SNOR 40 - validation during csv change

### CODE REVIEW

File: src/main/java/com/mycompany/myapp/service/CSV/CSVParser.java [hunk 1]
```diff
@@ -61,17 +66,13 @@
...
     }
 
-    private static Boolean verify(List<ParsingContainerDTO> parsedObjs, List<String> poolIdInDatabase) {
-        Map<String, ParsingContainerDTO> objects =
-            parsedObjs.toLinkedMap(parsingContainerDTO -> Tuple.of(parsingContainerDTO.getPoolId(),
-                parsingContainerDTO));
-        return poolIdInDatabase
-            .forAll(poolId -> !objects.containsKey(poolId));
+    private static Boolean verify(List<ParsingContainerDTO> parsedObjs) {
+        return parsedObjs.toJavaStream().allMatch(object -> object.getMaximumCount() >= reservationService.getActiveOrInFutureReservedCount(object.getPoolId()));
```
Comment (Pituuss) [inline src/main/java/com/mycompany/myapp/service/CSV/CSVParser.java]: It is a vavr list so you don't need to call `toJavaStream`, instead simply use `forAll` on that list

Comment (dwitek) [reply src/main/java/com/mycompany/myapp/service/CSV/CSVParser.java]: fixed

File: src/main/java/com/mycompany/myapp/service/reservation/ReservationService.java [hunk 2]
```diff
@@ -51,15 +50,27 @@ public void deleteReservation(Long id) {
 
     public List<Reservation> getAllByPoolId(String poolId) {
         Pool pool = poolRepository.getFullByIdWithReservation(poolId);
+        if (pool == null) {
```
Comment (Pituuss) [inline src/main/java/com/mycompany/myapp/service/reservation/ReservationService.java:53]: Is it even possible to get null there ?

Comment (dwitek) [reply src/main/java/com/mycompany/myapp/service/reservation/ReservationService.java:53]: Yes, when the pool doesn't exist yet in database, for example we upload updated csv with some new machines.

Comment (Pituuss) [reply src/main/java/com/mycompany/myapp/service/reservation/ReservationService.java:53]: ok then

**Pull request merged**